addons/website_blog/controllers/main.py

```diff
@@ -205,27 +188,21 @@ def blog_post(self, blog, blog_post, tag_id=None, page=1, enable_editor=None, **
         pager_end = page * self._post_comment_per_page
         comments = blog_post.website_message_ids[pager_begin:pager_end]
 
-        tag = None
-        if tag_id:
-            tag = request.registry['blog.tag'].browse(request.cr, request.uid, int(tag_id), context=request.context)
-        blog_url = QueryURL('', ['blog', 'tag'], blog=blog_post.blog_id, tag=tag, date_begin=date_begin, date_end=date_end)
+        blog_url = QueryURL('', ['blog', 'tag'], blog=blog_post.blog_id, tag=None, date_begin=date_begin, date_end=date_end)
 
         if not blog_post.blog_id.id == blog.id:
             return request.redirect("/blog/%s/post/%s" % (slug(blog_post.blog_id), slug(blog_post)))
 
-        tags = tag_obj.browse(cr, uid, tag_obj.search(cr, uid, [], context=context), context=context)
+        tags = request.env['blog.tag'].search([])
 
         # Find next Post
-        all_post_ids = blog_post_obj.search(cr, uid, [('blog_id', '=', blog.id)], context=context)
+        all_post_ids = request.env['blog.post'].search([('blog_id', '=', blog.id)])
```

Comment:
now search return recordset not ids. change variable name "all_posts" instead of "all_post_ids"